examples/suites/main_test.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package suites
+
+import (
+	"context"
+	"testing"
+
+	"sigs.k8s.io/e2e-framework/pkg/env"
+)
+
+var (
+	testenv env.Environment
+	ctx = context.Background()
+)
+
+func TestMain(m *testing.M) {
```

Comment:
What are the ordering guarentees between TestMain and other Test* functions?

Reply:
Or to be more exact `testenv.Run` vs `testenv.Test`

Reply:
Oh nvm, TIL `func TestMain(m *testing.M)` is from standard library

Reply:
I guess a follow-up question to this is if the TestMain pattern is required?

Reply:
@andrewsykim No, made sure that you don't have to start at the package level. So you don't have start with TestMain.  But if you do, you get wider lifecycle including callbacks prior to all tests started, and callback after all tests are done.